reattacher/reattacher_main.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+#include <getopt.h>
+#include "cclient/api/core/core_api.h"
+#include "cclient/api/extended/extended_api.h"
+#include "common/model/bundle.h"
+#include "storage/ta_storage.h"
+#include "time.h"
+
+#define PRESISTENT_PENDING_SECOND 1800 /**< 30 mins */
+#define REATTACH_MILESTONE_DEPTH 3
+#define REATTACH_MWM 14
+#define DELAY_INTERVAL 300 /**< 5 mins */
+
+#define logger_id scylladb_logger_id
+
+static status_t init_iota_client_service(iota_client_service_t* const serv) {
+  if (serv == NULL) {
+    ta_log_error("Invalid NULL pointer : serv");
```

Comment:
How about changing the error messages to `"Invalid NULL pointer specified by %s", __func__`?

Reply:
The logger macro has involved `__func__`.

```
#define ta_log_error(args...)                               \
  do {                                                      \
    log_error(logger_id, "[%s : %d] ", __func__, __LINE__); \
    log_error(logger_id, ##args);                           \
    fflush(stdout);                                         \
  } while (0)
```

Reply:
My bad. You don't have to mention parameter `serv` though.

Reply:
Got it.